src/apps/backend/modules/account/internal/account-reader.ts

```diff
@@ -53,15 +56,68 @@ export default class AccountReader {
     return AccountUtil.convertAccountDBToAccount(accDb);
   }
 
+  public static async getAccountByPhoneNumber(
+    phoneNumber: PhoneNumber,
+  ): Promise<Account> {
+    const accDb = await AccountRepository.findOne({
+      'phoneNumber.countryCode': phoneNumber.countryCode,
+      'phoneNumber.phoneNumber': phoneNumber.phoneNumber,
+      active: true,
+    });
+
+    const phoneNumberObj = new PhoneNumber(phoneNumber.countryCode, phoneNumber.phoneNumber);
+
+    if (!accDb) {
+      throw new AccountNotFoundError(phoneNumberObj.toString());
+    }
+
+    return AccountUtil.convertAccountDBToAccount(accDb);
+  }
+
+  public static async getAccountByPhoneNumberOptional(
+    phoneNumber: PhoneNumber,
+  ): Promise<Nullable<Account>> {
+    const accDb = await AccountRepository.findOne({
+      'phoneNumber.countryCode': phoneNumber.countryCode,
+      'phoneNumber.phoneNumber': phoneNumber.phoneNumber,
+      active: true,
+    });
+
+    if (!accDb) {
+      return null;
+    }
+
+    return AccountUtil.convertAccountDBToAccount(accDb);
+  }
+
   public static async checkUsernameNotExists(
-    params: AccountSearchParams,
+    username: string,
   ): Promise<boolean> {
     const accDb = await AccountRepository.findOne({
-      username: params.username,
+      username,
       active: true,
     });
+
+    if (accDb) {
+      throw new AccountWithUserNameExistsError(username);
+    }
+
+    return false;
+  }
+
+  public static async checkPhoneNumberNotExists(
+    phoneNumber: PhoneNumber,
+  ): Promise<boolean> {
+    const accDb = await AccountRepository.findOne({
+      'phoneNumber.countryCode': phoneNumber.countryCode,
+      'phoneNumber.phoneNumber': phoneNumber.phoneNumber,
+      active: true,
+    });
+
+    const phoneNumberObj = new PhoneNumber(phoneNumber.countryCode, phoneNumber.phoneNumber);
```

Comment:
This does not sound right.

phoneNumber is already being passed. If you have to do this, it tells me that the caller is not passing a real object of PhoneNumber to being with.